### PR TITLE
Grafana

### PR DESCRIPTION
**Pull request by wtedeschi**
Hi Kurt,

There we go with the Grafana support.

Best regards,
-Fred

### CODE REVIEW

File: src/stat.c [hunk 1]
```diff
@@ -134,29 +140,61 @@ static inline void StatUpdateMinMax(uint64_t newValue, uint64_t *min, uint64_t *
     *min = (newValue < *min) ? newValue : *min;
 }
 
+static void resetStats(struct Statistics *stats) {
```
Comment (shifty91) [inline src/stat.c:143]: StatResetStats()

File: src/stat.c [hunk 2]
```diff
@@ -134,29 +140,61 @@ static inline void StatUpdateMinMax(uint64_t newValue, uint64_t *min, uint64_t *
...
+            statPerPeriodPre->RoundTripOutliers++;
+        }
         stat->RoundTripCount++;
+        statPerPeriodPre->RoundTripCount++;
+
         stat->RoundTripSum += rtTime;
+        statPerPeriodPre->RoundTripSum += rtTime;
+
         stat->RoundTripAvg = stat->RoundTripSum / (double)stat->RoundTripCount;
+        statPerPeriodPre->RoundTripAvg = statPerPeriodPre->RoundTripSum / (double)statPerPeriodPre->RoundTripCount;
```
Comment (shifty91) [inline src/stat.c:197]: Is it possible to guard the whole `statPerPeriodPre` updates with `#if WITH_MQTT`, e.g., by moving it into a separate function?

**Review by shifty91 — CHANGES_REQUESTED** [review]
Thank you, Fred. Some minor comments. Do you mind adding documentation for the newly introduced configuration options, build options and dependencies?

File: app_config.in [hunk 3]
```diff
@@ -10,6 +10,7 @@
 #define VERSION "${VERSION}"
 #define INSTALL_EBPF_DIR "${INSTALL_EBPF_DIR}"
 
+#cmakedefine HAVE_SO_MOSQUITTO @HAVE_SO_MOSQUITTO@
```
[the remark this answers is not in the document]
Comment (wtedeschi) [reply app_config.in:13]: SO is trick, While looking at it i thought about shared object :D

File: src/logviamqtt.h [hunk 4]
```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) 2020-2022 Linutronix GmbH
+ * Author Walfred Tedeschi <walfred.tedeschi@intel.com>
```
[the remark this answers is not in the document]
Comment (wtedeschi) [reply src/logviamqtt.h:4]: I will fix it!

File: src/stat.c (hunk 2 above)
Comment (wtedeschi) [reply src/stat.c:197]: Makes sense as well!

File: src/logviamqtt.c [hunk 5]
```diff
@@ -0,0 +1,301 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) 2020-2022 Linutronix GmbH
+ * Author Walfred Tedeschi <walfred.tedeschi@intel.com>
```
[the remark this answers is not in the document]
Comment (wtedeschi) [reply src/logviamqtt.c:4]: Will fix it

File: src/logviamqtt.c [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+}
+
+void LogViaMQTTStats(enum StatFrameType frameType, struct Statistics *stats)
+{
+
+}
+#else
+
+/* 1 MiB / core */
+#define LOGVIAMQTT_BUFFER_SIZE (8 * 1024 * 1024)
```
[the remark this answers is not in the document]
Comment (wtedeschi) [reply src/logviamqtt.c:44]: I reduce by 1/4 but i don't think we need this amount of memory. However, i do think we need to set an inferior limit for the StatPeriod

[the remark this answers is not in the document]
Comment (wtedeschi) [reply src/logviamqtt.c:44]: At first we could set it limited to a min of 1s and max 10s

**Comment by wtedeschi** [comment]
One additional thing it would make sense to add a sample of Grafana setup, we could do RTT dashboard for the Real-time traffic

**Pull request closed without merging**